android/app/src/main/java/org/mozilla/magnet/NotificationService.java

```diff
@@ -0,0 +1,174 @@
+import android.content.Context;
+import android.content.Intent;
+import android.content.ServiceConnection;
+import android.content.SharedPreferences;
+import android.os.Handler;
+import android.os.IBinder;
+import android.support.v4.app.NotificationCompat;
+import android.util.Log;
+
+import org.mozilla.magnet.net.scanner.MagnetScannerItem;
+
+import java.util.HashMap;
+
+public class NotificationService extends Service implements ScannerService.ScannerServiceCallback {
+    private String TAG = "NotificationService";
+    private static final int SCAN_DURATION_MS = 5000;
+    public static final int NOTIFICATION_ID = 1;
+    private HashMap<String, MagnetScannerItem> mItems;
+    private ServiceConnection mServiceConnection;
+    private ScannerService mScannerService;
+    private boolean mConnected = false;
+
+    /**
+     * Called once, when the service is created.
+     */
+    @Override
+    public void onCreate() {
+        Log.d(TAG, "on create");
+
+        // background scanning isn't required
+        // if the app is 'active' (in foreground)
+        if (appActive()) {
+            stopSelf();
+            return;
+        }
+
+        connect();
+    }
+
+    /**
+     * Called once, when the service is terminated.
+     */
+    @Override
+    public void onDestroy() {
+        Log.d(TAG, "on destroy");
+        disconnect();
+    }
+
+    /**
+     * Connects to the service.
+     */
+    private void connect() {
+        mServiceConnection = new ServiceConnection() {
+
+            @Override
+            public void onServiceConnected(ComponentName className, IBinder service) {
+                Log.d(TAG, "on service connected");
+                mConnected = true;
+                ScannerService.LocalBinder binder = (ScannerService.LocalBinder) service;
+                mScannerService = binder.getService();
+                onConnected();
+            }
+
+            @Override
+            public void onServiceDisconnected(ComponentName arg0) {
+                Log.d(TAG, "on service disconnected");
+            }
+        };
+
+        Intent serviceIntent = new Intent(this, ScannerService.class);
+        bindService(serviceIntent, mServiceConnection, Context.BIND_AUTO_CREATE);
+    }
+
+    /**
+     * Cleans up and disconnects from the service.
+     */
+    private void disconnect() {
+        if (!mConnected) {
+            return;
+        }
+
+        mScannerService.removeListener(this);
+        mScannerService.stop();
+        unbindService(mServiceConnection);
+    }
+
+    /**
+     * Gets currently found items and listens for new items
+     * for a few seconds (SCAN_DURATION_MS). Once complete
+     * the service terminates itself, which in-turn triggers
+     * the cleanup in `onDestroy()`.
+     */
+    private void onConnected() {
+        mScannerService.addListener(NotificationService.this);
+        mItems = mScannerService.getItems();
+        updateNotification();
+
+        mScannerService.start();
```

Comment:
We call `updateNotifications()` here to update the state of current notification. This may result in a stale notification being removed or if we got some items back from the service the notification will be updated.

Then if/when we find items, we will update again.